test/accessor_tests.cc

```diff
@@ -684,5 +684,38 @@ namespace detail {
 		CHECK_THAT(lc->get_log(), Catch::Matchers::ContainsSubstring(error_message));
 	}
 
+	TEMPLATE_TEST_CASE_METHOD_SIG(oob_fixture, "host accessor reports out-of-bounds accesses", "[accessor][oob]", ((int Dims), Dims), 1, 2, 3) {
+#if !CELERITY_ACCESSOR_BOUNDARY_CHECK
+		SKIP("CELERITY_ACCESSOR_BOUNDARY_CHECK=0");
+#endif
+		buffer<int, Dims> buff(range_cast<Dims>(range<3>{10, 20, 30}));
+		const auto accessible_sr = subrange_cast<Dims>(subrange<3>{{5, 10, 15}, {1, 2, 3}});
+		const auto oob_idx_lo = id_cast<Dims>(id<3>{1, 2, 3});
+		const auto oob_idx_hi = id_cast<Dims>(id<3>{7, 13, 25});
+
+		// we need to be careful about the orderign of the construction and destruction
```

Comment:
"ordering" typo here (probably originally from me if this is copied, and also in the other test)